since v2.2.1 build failed on Alpine Linux

It was built successfully in previous versions.

```
/bin/sh ./version.sh
make  all-am
make[1]: Entering directory '/usr/local/src'
  CXX      memtier_benchmark-memtier_benchmark.o
memtier_benchmark.cpp:44:10: fatal error: execinfo.h: No such file or directory
   44 | #include <execinfo.h>
      |          ^~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:701: memtier_benchmark-memtier_benchmark.o] Error 1
make[1]: Leaving directory '/usr/local/src'
make: *** [Makefile:465: all] Error 2
```

[libexecinfo-dev](https://pkgs.alpinelinux.org/package/v3.16/main/x86_64/libexecinfo-dev) has been removed since Alpine v3.17

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -358,6 +358,38 @@ jobs:
         make code-coverage-capture
         bash <(curl -s https://codecov.io/bash) -f memtier_benchmark-*-coverage.info || echo "Codecov did not collect coverage reports"
 
+  build-alpine:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        alpine_version:
+          - "alpine:3.17"
+          - "alpine:3.18"
+          - "alpine:3.19"
+          - "alpine:3.20"
+          - "alpine:3.21"
+          - "alpine:latest"
+    container: ${{ matrix.alpine_version }}
+    name: Build ${{ matrix.alpine_version }}
+    steps:
+    - name: Install git
+      run: apk add --no-cache git
+
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Install build dependencies
+      run: |
+        apk add --no-cache \
+          make g++ autoconf automake libtool pkgconfig \
+          libevent-dev zlib-dev openssl-dev
+
+    - name: Build
+      run: autoreconf -ivf && ./configure && make -j
+
+    - name: Verify version
+      run: ./memtier_benchmark --version
+
   build-macos:
     strategy:
       matrix:

--- a/configure.ac
+++ b/configure.ac
@@ -37,6 +37,7 @@ AC_HEADER_STDC
 AC_HEADER_DIRENT
 AC_CHECK_HEADERS([stdlib.h string.h sys/time.h getopt.h limits.h malloc.h stdlib.h unistd.h utime.h assert.h sys/socket.h sys/types.h])
 AC_CHECK_HEADERS([fcntl.h netinet/tcp.h])
+AC_CHECK_HEADERS([execinfo.h])
 AC_CHECK_HEADERS([pthread.h])
 AC_CHECK_HEADERS([zlib.h])
 AC_CHECK_HEADERS([event2/event.h])

--- a/memtier_benchmark.cpp
+++ b/memtier_benchmark.cpp
@@ -41,7 +41,9 @@
 #include <sys/time.h>
 #include <sys/resource.h>
 #include <signal.h>
+#ifdef HAVE_EXECINFO_H
 #include <execinfo.h>
+#endif
 #include <ucontext.h>
 #include <time.h>
 #include <ctype.h>
@@ -1819,6 +1821,7 @@ static void print_all_threads_stack_trace(FILE *fp, int pid, const char *timestr
     // Print main/crashing thread first
     fprintf(fp, "[%d] %s # Thread %lu (current/crashing thread):\n", pid, timestr, (unsigned long) current_thread);
 
+#ifdef HAVE_EXECINFO_H
     void *trace[100];
     int trace_size = backtrace(trace, 100);
     char **messages = backtrace_symbols(trace, trace_size);
@@ -1826,6 +1829,9 @@ static void print_all_threads_stack_trace(FILE *fp, int pid, const char *timestr
         fprintf(fp, "[%d] %s #   %s\n", pid, timestr, messages[i]);
     }
     free(messages);
+#else
+    fprintf(fp, "[%d] %s #   (backtrace not available on this platform)\n", pid, timestr);
+#endif
 
     // Now print stack traces for worker threads if available
     if (g_threads != NULL) {

